### ISSUE TITLE
Color wise Component swizzle and function on vector types 

### ISSUE THREAD

**Issue by lieene** [Unity-Technologies/Unity.Mathematics#183]
For shader language compatibility and ease of use.
float4.rgba alike swizzle is expected.
And color-wise functions are expected too.
For example:
HsvToRgb()
RgbToHsv()
GammarToLinear()
LinearToGammar()

**Comment by unpacklo** [comment]
Yes, we've already been working on this internally to see if it can be done!

The main issues blocking us right now are that it actually requires Burst changes to support it and making sure that Mathematics compile time is not significantly affected. Because the swizzle permutation count is so large, we suspect that it's a significant contributor to long compile times and we don't want to make it longer.

We'll update this issue in the future when we have made a determination about whether or not this feature will be completed.
